app/src/test/java/tech/relaycorp/gateway/pdc/local/routes/ParcelCollectionRouteTest.kt

```diff
@@ -0,0 +1,195 @@
+                ) { incoming, _ ->
+                    delay(3_000) // wait to see if the connection is kept alive
+                    assertFalse(incoming.isClosedForReceive)
+                }
+            }
+        }
+
+    @Test
+    fun `Server should keep connection if Keep-Alive is invalid value`() =
+        runBlockingTest {
+            whenever(collectParcels.anyParcelsLeftToDeliverOrAck).thenReturn(flowOf(true))
+
+            testPDCServerRoute(route) {
+                handleWebSocketConversation(
+                    ParcelCollectionRoute.URL_PATH,
+                    { addHeader(ParcelCollectionRoute.HEADER_KEEP_ALIVE, "whatever") }
+                ) { incoming, _ ->
+                    delay(3_000) // wait to see if the connection is kept alive
+                    assertFalse(incoming.isClosedForReceive)
+                }
+            }
+        }
+
+    @Test
+    fun `Server should send parcel to deliver`() =
+        runBlockingTest {
+            val parcel = Pair("abc", ByteArray(0).inputStream())
+            whenever(collectParcels.getNewParcelsForEndpoints(any()))
+                .thenReturn(flowOf(listOf(parcel)))
+
+            testPDCServerRoute(route) {
+                handleWebSocketConversation(ParcelCollectionRoute.URL_PATH) { incoming, _ ->
+                    val parcelFrame = incoming.receive() as Frame.Binary
+                    val parcelDelivery = ParcelDelivery.deserialize(parcelFrame.data)
+                    assertEquals(parcel.first, parcelDelivery.deliveryId)
+                }
+            }
+        }
+
+    @Test
+    fun `Server should process client parcel acks`() =
+        runBlockingTest {
+            testPDCServerRoute(route) {
+                handleWebSocketConversation(ParcelCollectionRoute.URL_PATH) { _, outgoing ->
+                    val localId = "abc"
+
+                    outgoing.send(Frame.Text(localId))
+
+                    suspendWaitFor {
+                        verify(collectParcels).processParcelAck(eq(localId))
+                    }
+                }
+            }
+        }
+
+    @Test
+    fun `Server should ignore binary acks`() =
+        runBlockingTest {
+            testPDCServerRoute(route) {
+                handleWebSocketConversation(ParcelCollectionRoute.URL_PATH) { _, outgoing ->
+                    outgoing.send(Frame.Binary(true, "abc".toByteArray()))
+
+                    verify(collectParcels, never()).processParcelAck(any())
+                }
+            }
+        }
+
+    @Test
+    fun `Server should handle gracefully client abrupt close`() =
+        runBlockingTest {
+            testPDCServerRoute(route) {
+                handleWebSocketConversation(ParcelCollectionRoute.URL_PATH) { incoming, _ ->
+                    this@testPDCServerRoute.stop(0, 0)
+                    // TODO: figure out a way to tell if the server handled this correctly
+                }
+            }
+        }
+
```

Comment:
As mentioned on Slack, I think we should abandon this and undo the code to support it in the test suite and the UUT.